python/nav/portadmin/snmp/cisco.py

```diff
@@ -300,5 +309,59 @@ def get_dot1x_enabled_interfaces(self):
             for oid, state in self._bulkwalk(self.dot1xPortAuth)
         }
 
+    def get_poe_state_options(self):
+        """Returns the available options for enabling/disabling PoE on this netbox"""
+        options_list = [
+            self.POE_AUTO,
+            self.POE_STATIC,
+            self.POE_LIMIT,
+            self.POE_DISABLE,
+        ]
+        return tuple(options_list)
+
+    @translate_protocol_errors
+    def set_poe_state(self, interface, state):
+        unit_number, interface_number = self._get_poe_indexes_for_interface(interface)
+        oid_with_unit_number = self.POEENABLE + f".{unit_number}"
+        try:
+            self._set_netbox_value(
+                oid_with_unit_number, interface_number, 'i', state.state
+            )
+        except SnmpError as error:
+            _logger.error('Error setting poe state: %s', error)
+            raise
+        except ValueError as error:
+            _logger.error('%s is not a valid option for poe state', state)
+            raise
+
+    @translate_protocol_errors
+    def get_poe_state(self, interface):
+        unit_number, interface_number = self._get_poe_indexes_for_interface(interface)
+        oid_with_unit_number = self.POEENABLE + f".{unit_number}"
+        state_value = self._query_netbox(oid_with_unit_number, interface_number)
+        if state_value == None:
+            raise ManagementError("This interface does not support PoE")
+        for state in self.get_poe_state_options():
+            if state.state == state_value:
+                return state
+        raise ManagementError(f"Unknown PoE state {state_value}")
+
+    @translate_protocol_errors
+    def interface_supports_poe(self, interface):
+        try:
+            self.get_poe_state(interface)
+            return True
+        except ManagementError:
+            return False
+
+    def _get_poe_indexes_for_interface(self, interface):
+        try:
+            poeport = POEPort.objects.get(interface=interface)
+        except POEPort.DoesNotExist:
+            raise ManagementError("This interface does not have PoE indexes defined")
+        unit_number = poeport.poegroup.index
+        interface_number = poeport.index
+        return unit_number, interface_number
```

Comment:
Make sublasses of ManagementError and call those instead of using the one umbrella